extensions/cornerstone/src/services/ViewportService/CornerstoneViewportService.ts

```diff
@@ -506,19 +506,16 @@ class CornerstoneViewportService extends PubSubService
       !hangingProtocolService.customImageLoadPerformed
     ) {
       // delegate the volume loading to the hanging protocol service if it has a custom image load strategy
-      if (
-        hangingProtocolService.runImageLoadStrategy({
-          viewportId: viewport.id,
-          volumeInputArray,
-        })
-      ) {
-        // Fallback to the default strategy if the custom one fails
-        return;
-      }
+      return hangingProtocolService.runImageLoadStrategy({
+        viewportId: viewport.id,
+        volumeInputArray,
+      });
     }
 
     volumeToLoad.forEach(volume => {
-      volume.load();
+      if (!volume.loadStatus.loaded && !volume.loadStatus.loading) {
```

Comment:
What about cancelled?

Reply:
we should be able to resume the cancel, so I don't think so

Reply:
If so, you need to remove the cancelled flag in the load then - that can wait for now, but you might want to test it.